cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1717,11 +1770,14 @@ public List<ConfigMap> generatePerBrokerConfigurationConfigMaps(MetricsAndLoggin
                 // script to generate the node configuration.
                 data.put(BROKER_LISTENERS_FILENAME, node.broker() ? listeners.stream().map(ListenersUtils::envVarIdentifier).collect(Collectors.joining(" ")) : null);
 
-                if (useKRaft) {
+                // controller and broker gets the Cluster ID in different states during migration
+                // and they both get it when in full KRaft-mode
+                if (this.kafkaMetadataConfigState.isKRaftInConfiguration(node)) {
                     // In KRaft, we need to pass the Kafka CLuster ID and the metadata version
                     data.put(BROKER_CLUSTER_ID_FILENAME, clusterId);
                     data.put(BROKER_METADATA_VERSION_FILENAME, metadataVersion);
                 }
+                data.put("metadata.state", String.valueOf(this.kafkaMetadataConfigState.ordinal()));
```

Comment:
Should we have a constant for it as we have for all the other fields?

Reply:
Agreed. Not sure why I haven't created one.